File: src/main/java/com/testainers/SimpleHealthCheck.java
package com.testainers;

import jakarta.enterprise.context.ApplicationScoped;
import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;
import org.eclipse.microprofile.health.Liveness;

/**
 * @author Eduardo Folly
 */
@Liveness
@ApplicationScoped
public class SimpleHealthCheck implements HealthCheck {

    @Override
    public HealthCheckResponse call() {
        return HealthCheckResponse.up("httpbucket");
    }

}

File: src/main/kotlin/com/testainers/SimpleHealthCheck.kt
package com.testainers

import jakarta.enterprise.context.ApplicationScoped
import org.eclipse.microprofile.health.*

/**
 * @author Eduardo Folly
 */
@Liveness
@ApplicationScoped
class SimpleHealthCheck : HealthCheck {
    override fun call(): HealthCheckResponse =
        HealthCheckResponse.up("httpbucket")
}